pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -271,6 +271,46 @@ impl<T: Config> Pallet<T> {
 		fee
 	}
 
+	/// Compute the length portion of a fee by invoking the configured `LengthToFee` impl.
+	/// # Arguments
+	/// * `length` - The length of the transaction.
+	/// * `weight` - The weight of the transaction.
+	///
+	/// # Returns
+	/// `FeeDetails` - The fee details for the transaction.
+	pub fn compute_capacity_fee_details(
+		runtime_call: &<T as Config>::RuntimeCall,
+		dispatch_weight: &Weight,
+		len: u32,
+	) -> FeeDetails<BalanceOf<T>> {
+		let calls = T::CapacityCalls::get_inner_calls(runtime_call)
+			.expect("A collection of calls is expected at minimum one.");
+
+		let mut calls_weight_sum = Weight::zero();
+		for inner_call in calls {
+			let call_weight = T::CapacityCalls::get_stable_weight(&inner_call).unwrap_or_default();
+			calls_weight_sum = calls_weight_sum.saturating_add(call_weight);
+		}
+
+		let fees = FeeDetails { inclusion_fee: None, tip: Zero::zero() };
+		if calls_weight_sum.is_zero() {
```

Comment:
If the call_weigh_sum is zero but the dispatch calls something, do we want to return the cost of the dispatch call?

Reply:
That might give a false idea that the call they sent is capacity enabled or not, but open to suggestions

Reply:
small suggestion. Since we are returning here maybe get rid of the else if:
```
if { ... return}
if let Some(weight) = calll... {
}
else {
}
```

Reply:
yep cleaned it up